src/libfuncs.rs

```diff
@@ -507,3 +507,258 @@ pub fn increment_builtin_counter<'ctx: 'a, 'a>(
         .result(0)?
         .into())
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use melior::dialect::DialectRegistry;
+    use melior::ir::Type;
+    use melior::utility::register_all_dialects;
+
+    /// Load all available MLIR dialects into the given context.
+    ///
+    /// This function initializes a new `DialectRegistry`, registers all available dialects
+    /// into it, appends the registry to the provided context, and then loads all available
+    /// dialects into the context.
+    pub fn load_all_dialects(context: &Context) {
+        let registry = DialectRegistry::new();
+        register_all_dialects(&registry);
+        context.append_dialect_registry(&registry);
+        context.load_all_available_dialects();
+    }
+
+    #[test]
+    fn switch_branch_arg_external_test() {
+        // Create a new context for MLIR operations
+        let context = Context::new();
+        // Load all available MLIR dialects into the context
+        load_all_dialects(&context);
+
+        // Create an unknown location in the context
+        let location = Location::unknown(&context);
+        // Create a new MLIR module with the unknown location
+        let module = Module::new(location);
+
+        // Create a new MLIR block and obtain its reference
+        let last_block = unsafe { &BlockRef::from_raw(Block::new(&[]).into_raw()) };
```

Comment:
You dont need unsafe to get a BlockRef.

Also first you need to create a function with the `func` dialect, check out the melior crate example. There you will need to add a region, and you add blocks to the region, when adding the block to the region you will get a BlockRef

Reply:
Nice, just fixed. As I can have an empty block here, I didn't add a function inside it as you pointed out inside your comment. I think it's useless in this situation, but I may be wrong. Let me know if it's ok for you.

Reply:
You are right about not needing the function.